### PR TITLE
Allow passing JSON file to checkver

### PR DESCRIPTION
**Pull request by r15ch13**
- Moved JSON parsing into the queue
  - Removes some loops
- Allow passing a JSON file path to `checkver.ps1`
  - Example: `> .\bin\checkver.ps1 -App ..\somewhere\bucket\test.json -Force`
- Use proper parameters values for `next()`
- Improve event cleanup

### CODE REVIEW

**Comment by r15ch13** [comment]
ping @Ash258 @niheaven

**Comment by Ash258** [comment]
Looks good by fast static review.

I do not know when i will find time to do proper review. I guess Friday / weekend evenings should be good.

File: bin/checkver.ps1 [hunk 1]
```diff
@@ -72,27 +72,28 @@ param(
...
 # get apps to check
 $Queue = @()
-$json = ''
-Get-ChildItem $Dir "$App.json" | ForEach-Object {
-    $json = parse_json "$Dir\$($_.Name)"
-    if ($json.checkver) {
-        $Queue += , @($_.Name, $json)
-    }
+if((Test-Path $App) -and $App -ne "*") {
+    $Queue = @(Get-ChildItem $App)
```
Comment (niheaven) [inline bin/checkver.ps1]:
```suggestion
    $Queue = @(Get-Item -Path (Convert-Path -Path $App))
```

File: bin/checkver.ps1 [hunk 2]
```diff
@@ -72,27 +72,28 @@ param(
...
+} else {
+    $Dir = Resolve-Path $Dir
+    $Queue = @(Get-ChildItem -Path $Dir -Filter "$App.json")
```
Comment (niheaven) [inline bin/checkver.ps1]:
```suggestion
    $Queue = @(Get-ChildItem -Path (Convert-Path -Path $Dir) -Filter "$App.json")
```

Also remove above `$Dir = Resolve-Path $Dir`

**Comment by rashil2000** [comment]
Superceded by #5019

**Pull request closed without merging**